### PR TITLE
The event options added to the config

### PR DESCRIPTION
**Pull request by 2403905**
## Description

The `values-save` reqest
```curl
curl -XPOST --location 'https://localhost:9200/api/v0/settings/values-save'  -vk -uadmin:admin \
--header 'Content-Type: application/json' \
--data '{
    "value": {
        "id": "4e454222-4880-4899-8418-315f8aa284d1",
        "accountUuid": "me",
        "bundleId": "2a506de7-99bd-4f0d-994e-c38e72c28fd9",
        "settingId": "872d8ef6-6f2a-42ab-af7d-f53cc81d7046",
        "resource": {
            "type": "TYPE_USER"
        },
        "collectionValue": {
            "values": [
                {
                    "key": "in-app",
                    "boolValue": false
                },
                {
                    "key": "mail",
                    "boolValue": false
                }
            ]
        }
    }
}'
```

`https://localhost:9200/api/v0/settings/bundles-list`
```json
{
  "bundles": [
    {
      "id": "2a506de7-99bd-4f0d-994e-c38e72c28fd9",
      "name": "profile",
      "type": "TYPE_DEFAULT",
      "extension": "ocis-accounts",
      "displayName": "Profile",
      "settings": [
       ......

        {
          "id": "08dec2fe-3f97-42a9-9d1b-500855e92f25",
          "name": "email-sending-interval-options",
          "displayName": "Email Notifications options",
          "description": "Email notifications options",
          "singleChoiceValue": {
            "options": [
              {
                "value": {
                  "stringValue": "daily"
                },
                "displayValue": "Daily"
              },
              {
                "value": {
                  "stringValue": "weekly"
                },
                "displayValue": "Weekly"
              },
              {
                "value": {
                  "stringValue": "instant"
                },
                "displayValue": "Instant"
              },
              {
                "value": {
                  "stringValue": "never"
                },
                "displayValue": "Never"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "872d8ef6-6f2a-42ab-af7d-f53cc81d7046",
          "name": "event-share-created-options",
          "displayName": "Share created",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "d7484394-8321-4c84-9677-741ba71e1f80",
          "name": "event-share-removed-options",
          "displayName": "Share removed",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {}
                },
                "key": "mail",
                "attribute": "disabled",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "e1aa0b7c-1b0f-4072-9325-c643c89fee4e",
          "name": "event-share-expired-options",
          "displayName": "Share expired",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "694d5ee1-a41c-448c-8d14-396b95d2a918",
          "name": "event-space-shared-options",
          "displayName": "Space shared",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "26c20e0e-98df-4483-8a77-759b3a766af0",
          "name": "event-space-unshared-options",
          "displayName": "Space unshared",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "7275921e-b737-4074-ba91-3c2983be3edd",
          "name": "event-space-membership-expired-options",
          "displayName": "Space membership expired",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "eb5c716e-03be-42c6-9ed1-1105d24e109f",
          "name": "event-space-disabled-options",
          "displayName": "Space disabled",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {}
                },
                "key": "mail",
                "attribute": "disabled",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "094ceca9-5a00-40ba-bb1a-bbc7bccd39ee",
          "name": "event-space-deleted-options",
          "displayName": "Space deleted",
          "description": "Send In-App",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {}
                },
                "key": "mail",
                "attribute": "disabled",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "fe0a3011-d886-49c8-b797-33d02fa426ef",
          "name": "event-postprocessing-step-finished-options",
          "displayName": "Postprocessing Step Finished",
          "description": "Postprocessing Step Finished",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {}
                },
                "key": "mail",
                "attribute": "disabled",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        },
        {
          "id": "b441ffb1-f5ee-4733-a08f-48d03f6e7f22",
          "name": "event-Science-mesh-invite-token-generated-options",
          "displayName": "Science Mesh Invite Token Generated",
          "description": "Science Mesh Invite Token Generated",
          "multiChoiceCollectionValue": {
            "options": [
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "in-app",
                "displayValue": "In-App"
              },
              {
                "value": {
                  "boolValue": {
                    "default": true
                  }
                },
                "key": "mail",
                "displayValue": "Mail"
              }
            ]
          },
          "resource": {
            "type": "TYPE_USER"
          }
        }
      ],
      "resource": {
        "type": "TYPE_SYSTEM"
      }
    }
  ]
}

```

## Related Issue

- Related Issue [#10669](https://github.com/owncloud/ocis/issues/10669)

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by 2403905** [comment]
@micbar Can I merge these changes before release?

File: services/settings/pkg/service/v0/servicedecorator.go [hunk 1]
```diff
@@ -82,3 +94,93 @@ func (s *defaultLanguageDecorator) withDefaultLanguageSetting(accountUUID string
...
+}
+
+func (s *defaultLanguageDecorator) withDefaultProfileValueList(ctx context.Context,
+	accountUUID string, requested map[string]*settingsmsg.ValueWithIdentifier) map[string]*settingsmsg.ValueWithIdentifier {
+
+	// we use the default profile bundle instead of s.GetBundle(ctx, req, resp)
+	bundle := defaults.GenerateDefaultProfileBundle()
+
+	for _, setting := range bundle.GetSettings() {
+		if v, ok := requested[setting.GetId()]; ok && v == nil {
```
Comment (kobergj) [inline services/settings/pkg/service/v0/servicedecorator.go:116]: Can we reverse the logic here? Too much intendation :)
```suggestion
		if v, ok := requested[setting.GetId()]; !ok || v != nil {
		    continue
		}
```

File: services/settings/pkg/store/defaults/defaults.go [hunk 2]
```diff
@@ -94,6 +122,17 @@ func generateBundleAdminRole() *settingsmsg.Bundle {
...
+			ProfileEventShareCreatedPermission(Own),
+			ProfileEventShareRemovedPermission(Own),
+			ProfileEventShareExpiredPermission(Own),
+			ProfileEventSpaceSharedPermission(Own),
+			ProfileEventSpaceUnsharedPermission(Own),
+			ProfileEventSpaceMembershipExpiredPermission(Own),
+			ProfileEventSpaceDisabledPermission(Own),
+			ProfileEventSpaceDeletedPermission(Own),
+			ProfileEventPostprocessingStepFinishedPermission(Own),
+			ProfileEventScienceMeshInviteTokenGeneratedPermission(Own),
```
Comment (kobergj) [inline services/settings/pkg/store/defaults/defaults.go:135]: Question: Do we need a permission for every notification setting? Can't we just have one permission that allows changing all of them?

**Review by kobergj — CHANGES_REQUESTED** [review]
some nitpicks

File: services/settings/pkg/store/defaults/defaults.go [hunk 3]
```diff
@@ -24,6 +24,29 @@ const (
...
+	// SettingUUIDProfileEventSpaceMembershipExpired is the hardcoded setting UUID for the send in app setting
+	SettingUUIDProfileEventSpaceMembershipExpired = "7275921e-b737-4074-ba91-3c2983be3edd"
+	// SettingUUIDProfileEventSpaceDisabled is the hardcoded setting UUID for the send in app setting
+	SettingUUIDProfileEventSpaceDisabled = "eb5c716e-03be-42c6-9ed1-1105d24e109f"
+	// SettingUUIDProfileEventSpaceDeleted is the hardcoded setting UUID for the send in app setting
+	SettingUUIDProfileEventSpaceDeleted = "094ceca9-5a00-40ba-bb1a-bbc7bccd39ee"
+	// SettingUUIDProfileEventPostprocessingStepFinished is the hardcoded setting UUID for the send in mail setting
+	SettingUUIDProfileEventPostprocessingStepFinished = "fe0a3011-d886-49c8-b797-33d02fa426ef"
+	// SettingUUIDProfileEventScienceMeshInviteTokenGenerated is the hardcoded setting UUID for the send in mail setting
+	SettingUUIDProfileEventScienceMeshInviteTokenGenerated = "b441ffb1-f5ee-4733-a08f-48d03f6e7f22"
```
Comment (kobergj) [inline services/settings/pkg/store/defaults/defaults.go:49]: Just discussed with @bastianbeier:

We don't need a setting to switch of ScienceMeshInvitationToken emails. It should always be active as it makes no sense to turn that notification of.

**Review by kobergj — CHANGES_REQUESTED** [review]

File: changelog/unreleased/enhancement-add-events-setting.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Enhancement: The user settings was extended by the notifications preferences
+
+The user settings was extended, that a user can store their notifications preferences
+
+https://github.com/owncloud/ocis/pull/10706  
+https://github.com/owncloud/ocis/issues/10669
```
Comment (mmattel) [inline changelog/unreleased/enhancement-add-events-setting.md:6]:
```suggestion
Enhancement: Part I: The user settings are extended by notifications preferences

Part I: Users can now store their notifications preferences in the user settings

https://github.com/owncloud/ocis/pull/10706  
https://github.com/owncloud/ocis/issues/10669
```
This is the same changelog text as in #10779 - which I just reviewed (created two suggestable commits)
I do not think that this is a good idea to have two identical changelog entries.
This small change makes this distinguishable but anyone can see it belongs together (part I / part II)

File: services/settings/pkg/store/defaults/defaults.go (hunk 2 above)
Comment (2403905) [reply services/settings/pkg/store/defaults/defaults.go:135]: It is not possible now

**Pull request merged**